internal/k8s/validator.go

```diff
@@ -192,6 +194,23 @@ func (l *leader) ValidateHostRuleObj(key string, hostrule *akov1alpha1.HostRule)
 	return nil
 }
 
+func validateSecretReferenceInHostrule(namespace, secretName string) error {
+
+	// reject the hostrule if the secret handling is restricted to the namespace where
+	// AKO is installed.
+	if utils.GetInformers().RouteInformer != nil &&
+		namespace != utils.GetAKONamespace() &&
+		utils.IsSecretsHandlingRestrictedToAKONS() {
+		err := fmt.Errorf("secret handling is restricted to %s namespace only", utils.GetAKONamespace())
+		return err
+	}
+	_, err := utils.GetInformers().SecretInformer.Lister().Secrets(namespace).Get(secretName)
+	if err != nil {
```

Comment:
can we just return err here. `err` will be either nil or some value.

Reply:
I have addressed the comment.